common/achievement/helper.ts

```diff
@@ -0,0 +1,22 @@
+import { ActionID } from '../notification/actions';
+import { prisma } from '../prisma';
+import { metrics } from './metrics';
+
+function isMetricExistingForActionId<ID extends ActionID>(actionId: ID): boolean {
+    return metrics.some((m) => m.onActions.includes(actionId));
+}
+
+async function isAchievementExistingForAction<ID extends ActionID>(actionId: ID): Promise<boolean> {
+    const metricsByAction = metrics.filter((m) => m.onActions.includes(actionId));
+
+    for (const metric of metricsByAction) {
+        const templates = await prisma.achievement_template.findMany({ where: { metrics: { has: metric.metricName } } });
```

Comment:
It might be worth caching this in a Map per metric (like the Notification System does)